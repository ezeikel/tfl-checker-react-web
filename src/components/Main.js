import React from "react";
import PropTypes from "prop-types";
import { Switch, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import styled from "styled-components";
import Status from "./Status";
import TripPlanner from "./TripPlanner";
import Trip from "./Trip";

const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  padding: var(--spacing-large);
`;

const Main = ({ selectedTrip }) => (
  <Wrapper>
    <Switch>
      <Route exact path="/">
        <Status />
      </Route>
      <Route path="/trip-planner">
        <TripPlanner />
      </Route>
      <Route path="/trip">
        {selectedTrip ? <Trip /> : <Redirect to="/trip-planner" />}
      </Route>
    </Switch>
  </Wrapper>
);

const mapStateToProps = ({ suggestion }) => ({
  selectedTrip: suggestion.selected,
});

Main.defaultProps = {
  selectedTrip: {},
};

Main.propTypes = {
  selectedTrip: PropTypes.shape({
    duration: PropTypes.number,
    startDateTime: PropTypes.string,
    arrivalDateTime: PropTypes.string,
    legs: PropTypes.array,
  }),
};

export default connect(mapStateToProps)(Main);
