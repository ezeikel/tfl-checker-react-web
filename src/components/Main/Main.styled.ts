/* eslint-disable import/prefer-default-export */

import styled from "styled-components";

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  padding: var(--spacing-large);
`;
