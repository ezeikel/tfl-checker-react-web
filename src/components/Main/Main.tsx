import { Routes, Route } from "react-router-dom";
import Status from "../Status/Status";
import TripPlanner from "../TripPlanner/TripPlanner";
import Trip from "../Trip/Trip";
import { Wrapper } from "./Main.styled";

const Main = () => (
  <Wrapper>
    <Routes>
      <Route path="/" element={<Status />} />
      <Route path="/trip-planner" element={<TripPlanner />} />
      <Route path="/trip" element={<Trip />} />
    </Routes>
  </Wrapper>
);

export default Main;
