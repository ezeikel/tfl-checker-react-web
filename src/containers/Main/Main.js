import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Status from '../../components/Status/Status';
import Planner from '../../components/Planner/Planner';

const Main = () => (
    <main>
        <Switch>
            <Route exact path='/' component={Status} />
            <Route exact path='/planner' component={Planner} />
        </Switch>
    </main>
)

export default Main;
