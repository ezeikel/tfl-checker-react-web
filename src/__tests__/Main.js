import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { BrowserRouter as Router } from 'react-router-dom';
import Main from '../components/Main';

const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);

test('renders without crashing', () => {
  const store = mockStore({
    suggestion: {
      selected: {
        duration: 200,
        startDateTime: "",
        arrivalDateTime: "",
        legs: [],
      }
    }
  });
  const { container } = render(
    <Router>
      <Main store={store} /> { /** eslint-disable-line react/jsx-props-no-spreading */ }
    </Router>
  );
  expect(container).toMatchSnapshot();
});