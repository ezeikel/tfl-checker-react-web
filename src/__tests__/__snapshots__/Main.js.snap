// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <main
    class="sc-fzqAbL emdVaB"
  >
    <div
      class="sc-AxhCb kicRlZ"
    />
  </main>
   
</div>
`;
